src/axom/core/Array.hpp

```diff
@@ -136,13 +140,32 @@ class Array : public ArrayBase<T, DIM, Array<T, DIM, SPACE>>
         IndexType capacity = 0,
         int allocator_id = axom::detail::getAllocatorID<SPACE>());
 
+  /// \overload
+  template <IndexType SFINAE_DIM = DIM,
+            MemorySpace SFINAE_SPACE = SPACE,
+            typename std::enable_if<SFINAE_DIM == 1 &&
+                                    SFINAE_SPACE == MemorySpace::Dynamic>::type* = nullptr>
+  Array(ArrayOptions::Uninitialized,
+        IndexType num_elements,
+        IndexType capacity = 0,
+        int allocator_id = axom::detail::getAllocatorID<SPACE>());
```

Comment:
Note: `axom::Array` supports two features that can only be enabled through its constructors -- uninitialized data (via the `ArrayOptions::Uninitialized` tag) and a custom allocator (via the `allocatorID` argument) -- so I needed to add some extra overloads to support this.

I also improved the testing of this feature and ensured that when you copy and array with these features, the copy gets them as well.

Reply:
So the default behavior is to initialize the data and a user can choose to leave it uninitialized via an arg to the ctor?

Reply:
We did run into some issues with `axom::Array`s of non-trivial type, e.g. `axom::Array<BitSet>` in ImplicitGrid, from the array data being uninitialized - e.g. assigning a new bitset like `array[i] = BitsetType{...}` would call the destructor on junk data, which led to some weird segfaults.

I think making uninitialized data opt-in also keeps `axom::Array` more drop-in compatible with `std::vector`, since vector's behavior is to default-init all elements.

Reply:
@kanye-quest I agree 100%

Reply:
@kanye-quest -- is there a test for this?

I added this because, `Delaunay::isValid()` computation is using large bitsets, e.g. w/ millions of bit ( one per tetrahedron) and in `ImplicitGrid::findCandidates()`, the cost of initializing these bits to immediately get overwritten was substantial. I saw a substantial speedup when I disabled the initialization.